classes/export_webservices.php

```diff
@@ -0,0 +1,387 @@
+    public function __construct(string $host, array $selectedwebserviceindices) {
+        $this->host        = $host;
+        $this->webservices = $this->get_selected_webservice_objects($selectedwebserviceindices);
+    }
+
+    /**
+     * Exports data as cURL commands in a text file.
+     * Sets header to initiate download with filename and extension.
+     */
+    public function export_as_curl(): void {
+        header('Content-Disposition: attachment; filename=curl.txt');
+        header('Content-Type: application/plain');
+
+        $curlstrings = [];
+        foreach ($this->webservices as $webservice) {
+            $paramsarray = $this->get_formatted_param_array($webservice);
+
+            $curlstrings[] = 'curl ' . $this->create_request_string($webservice, $paramsarray);
+        }
+
+        foreach ($curlstrings as $curlstring) {
+            echo $curlstring . "\n" . "\n";
+        }
+    }
+
+
+    /**
+     * Exports data as Postman Collection in a json file.
+     * Sets header to initiate download with filename and extension.
+     */
+    public function export_as_postman(): void {
+        header('Content-Disposition: attachment; filename=postman.json');
+        header('Content-Type: application/json');
+
+        $postmanitems = [];
+        foreach ($this->webservices as $webservice) {
+            $paramsarray = $this->get_formatted_param_array($webservice);
+
+            $postmanitems[] = $this->create_postman_request_item($webservice, $paramsarray);
+        }
+
+        $postmancollection = $this->create_postman_collection($postmanitems);
+        $beautifiedjson    = json_encode($postmancollection, JSON_PRETTY_PRINT);
+        echo $beautifiedjson;
+    }
+
+
+    /**
+     * Retrieves an array of webservice objects based on provided indices.
+     *
+     * @param  array $selectedwebserviceindices An array of indices to grab the webservice objects.
+     * @return array An array of webservice objects.
+     */
+    private function get_selected_webservice_objects(array $selectedwebserviceindices): array {
+        $webservicesrecords = $this->get_indexed_webservice_records();
+
+        $webservices = [];
+        foreach ($selectedwebserviceindices as $index) {
+            $webservice    = $webservicesrecords[$index];
+            $webservices[] = external_api::external_function_info($webservice);
+        }
+
+        return $webservices;
+
+    }
+
+
+    /**
+     * Retrieves indexed webservice records from the database.
+     *
+     * @return array An array of the indexed webservices.
+     */
+    private function get_indexed_webservice_records(): array {
+        global $DB;
+
+        // Get_records returns an object array where key for each object is the name of the webservice.
+        // Use array_values to change key to the index of each object so that we can filter based on $selectedWebserviceIndices.
+        $webservicesrecords = array_values($DB->get_records('external_functions', [], ''));
```

Comment:
as a quick shortcut here (I dont think this is noted in the API docs), the params array defaults to [], and same with all other params, so to get all you can just call `$DB->get_records($table);`